python/servo/package_commands.py

```diff
@@ -71,6 +82,71 @@ def package(self, release=False, dev=False, android=None, debug=False, debugger=
             except subprocess.CalledProcessError as e:
                 print("Packaging Android exited with return value %d" % e.returncode)
                 return e.returncode
+        elif is_macosx():
+            dir_to_build = '/'.join(binary_path.split('/')[:-1])
+            dir_to_dmg = '/'.join(binary_path.split('/')[:-2]) + '/dmg'
+            dir_to_app = dir_to_dmg + '/Servo.app'
+            dir_to_root = '/'.join(binary_path.split('/')[:-3])
+            if os.path.exists(dir_to_dmg):
+                print("Cleaning up from previous packaging")
+                delete(dir_to_dmg)
+            browserhtml_path = find_dep_path_newest('browserhtml', binary_path)
+            new_browserhtml_path = dir_to_app + '/Contents/Resources/' + browserhtml_path.split('/')[-1]
+            if browserhtml_path is None:
+                print("Could not find browserhtml package; perhaps you haven't built Servo.")
+                return 1
+
+            print("Copying files")
+            shutil.copytree(dir_to_root + '/resources', dir_to_app + '/Contents/Resources')
+            shutil.copytree(browserhtml_path, new_browserhtml_path)
+            shutil.copy2(dir_to_root + '/Info.plist', dir_to_app + '/Contents/Info.plist')
+            os.makedirs(dir_to_app + '/Contents/MacOS/')
+            shutil.copy2(dir_to_build + '/servo', dir_to_app + '/Contents/MacOS/')
+
+            print("Finding dylibs to be copied")
+            need = set([dir_to_app + '/Contents/MacOS/servo'])
+            done = set()
+
+            while need:
+                needed = set(need)
+                need = set()
+                for f in needed:
+                    need.update(otool(f))
+                done.update(needed)
+                need.difference_update(done)
+
+            print("Copying dylibs")
+            for f in sorted(done):
+                if '/System/Library' not in f and '/usr/lib' not in f and 'servo' not in f:
+                    shutil.copyfile(f, dir_to_app + '/Contents/MacOS/' + f.split('/')[-1])
+                    install_name_tool(f, f.split('/')[-1], dir_to_app + '/Contents/MacOS/servo')
+
+            print("Writing run-servo")
+            # TODO: deduplicate this arg list from post_build_commands
+            servo_args = ['-w', '-b',
+                          '--pref', 'dom.mozbrowser.enabled',
+                          '--pref', 'dom.forcetouch.enabled',
+                          '--pref', 'shell.builtin-key-shortcuts.enabled=false',
+                          '--resources-path', '${0%/*}/../Resources',
+                          path.join('${0%/*}/../Resources', browserhtml_path.split('/')[-1], 'out', 'index.html')]
```

Comment:
We might have to keep the script just to get `shell.homepage` to work. Other than that, the prefs can be kept in a separate json.